pandas/core/indexes/range.py

```diff
@@ -592,26 +592,27 @@ def _evaluate_numeric_binop(self, other):
                 attrs = self._get_attributes_dict()
                 attrs = self._maybe_update_attributes(attrs)
 
+                left, right = self, other
                 if reversed:
-                    self, other = other, self
+                    left, right = right, left
 
                 try:
                     # apply if we have an override
                     if step:
                         with np.errstate(all='ignore'):
-                            rstep = step(self._step, other)
+                            rstep = step(left._step, right)
```

Comment:
Side-note: This is still not great; `left` may no longer be `self`, so this may raise an AttributeError.  The existing method specifically catches AttributeError so I guess this is intentional.

Reply:
use getattr

Reply:
Huh?

Reply:
``getattr(left, '_step', left)`` and remove the AttributeError catching

Reply:
``getattr(left, '_stop', left)``

Reply:
and remove the AttributeError catching

Reply:
can you do this

Reply:
Out of scope.  Until I look at this closely, I'm assuming that the author had a reason for doing it this way.

Reply:
ok fair enough